### PR TITLE
[Paywalls V2] Stack background, borders and shadows update depending on the theme

### PR DESCRIPTION
**Pull request by JayShortway**
## Description
As the title says. This also means that `StackComponentStyle` no longer needs a Composable context to be created, save for the recursive `StyleFactory.create()` call. Will be giving `TextComponentStyle` the same treatment in future PRs. This will allow us to move the `StyleFactory.create()` call to the ViewModel, and do validation in the `Loading` state. 

## Main changes
- Adds `StackComponentViewTests`.
- Uses the existing `rememberColorStyle()`.
- Adds and uses a new `rememberBorderStyle()`.
- Adds and uses a new `rememberShadowStyle()`.

## Other changes
- Adds dark mode variants of `StackComponentView` previews.
- Some convenience updates to `assertPixelColorEquals.kt`.

### CODE REVIEW

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Border.kt [hunk 1]
```diff
@@ -117,27 +118,29 @@ private fun Border_Preview_LinearGradient(shape: Shape) {
             .requiredSize(100.dp)
```
Comment (JayShortway) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Border.kt:118]: Best to review this file hiding whitespace.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Shadow.kt [hunk 2]
```diff
@@ -87,14 +87,16 @@ private fun Shadow_Preview_Circle() {
             modifier = Modifier
```
Comment (JayShortway) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Shadow.kt:87]: Best to review this file hiding whitespace.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/StackComponentView.kt [hunk 3]
```diff
@@ -53,14 +55,18 @@ internal fun StackComponentView(
...
         val commonModifier = remember(style) {
             Modifier
                 .padding(style.margin)
-                .applyIfNotNull(style.shadow) { shadow(it, style.shape) }
-                .applyIfNotNull(style.background) { background(it, style.shape) }
+                .applyIfNotNull(shadowStyle) { shadow(it, style.shape) }
+                .applyIfNotNull(backgroundColorStyle) { background(it, style.shape) }
                 .clip(style.shape)
-                .applyIfNotNull(style.border) { border(it, style.shape) }
+                .applyIfNotNull(borderStyle) { border(it, style.shape) }
```
Comment (JayShortway) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/StackComponentView.kt:69]: These are the actual fixes.

**Review by tonidero — COMMENTED** [review]
Changes look good but I'm actually confused... Why have there been changes in the non-component paywalls? Seems some slight changes on the fonts so maybe one of the base branches?

**Comment by JayShortway** [comment]
Yea those are unfortunately caused by the base branch, this PR: https://github.com/RevenueCat/purchases-android/pull/1981.

**Comment by JayShortway** [comment]
I'm not entirely sure why Emerge still thinks there are differences compared to `main`. Maybe it doesn't properly update when the target branch of the PR changes? I did re-run Emerge. At least there are no more changes to existing (V1) snapshots. I expect/hope this will right itself after merging.

**Review by tonidero — APPROVED** [review]

**Pull request merged**